src/index.ts

```diff
@@ -126,87 +41,27 @@ export class CalimeroSdk {
 
   signMessage = () => {
     const message =  uuidv4().toString();
-    localStorage.setItem(MESSAGE_KEY,
-      message);
     localStorage.setItem(
       MESSAGE_HASH_KEY,
       sha256.update(message ).toString()
     );
-    const callbackUrl = encodeURIComponent(window.location.href);
+    //to do -> change http://localhost:3000 to calimeroUrl from calimeroSdK config file
+    // eslint-disable-next-line max-len
+    const callbackUrl = encodeURIComponent(`http://localhost:3000/accounts/sync/?shard=${this._config.shardId}&next=${window.location.href}&ogm=${message}`);
     window.location.href =
         // eslint-disable-next-line max-len
         `${this._config.walletUrl}/verify-owner?message=${localStorage.getItem(MESSAGE_HASH_KEY)}&callbackUrl=${callbackUrl}`;
   }
 
-  authenticate = () => {
-    const search = window.location.search;
-    const params = new URLSearchParams(search);
-    const accountId = params.get("accountId");
-    const blockId = params.get("blockId");
-    const publicKey = params.get("publicKey");
-    const signature = params.get("signature");
-
-    if(!accountId || !blockId || !publicKey || !signature) {
-      clearLocalStorage();
-      console.error("Missing params");
-      return;
-    }
-    axios.post(
-      this._config.calimeroUrl + "/api/v1/authenticate",
-      {
-        shardId: this._config.shardId,
-        accountId,
-        blockId,
-        publicKey,
-        signature,
-        calimeroSecret: localStorage.getItem(MESSAGE_KEY),
-        calimeroSecretHash: localStorage.getItem(MESSAGE_HASH_KEY)
-      }
-    ).then((res: any) => {
-      if (res.status == 200) {
-        console.log(res.data);
-        localStorage.setItem(AUTH_TOKEN_KEY,
-          res.data.secretToken);
-        console.log(JSON.stringify(res.data.calimeroToken));
-        localStorage.setItem(CALIMERO_TOKEN_KEY,
-          JSON.stringify(res.data.calimeroToken));
-        window.location.reload();
-      }
-    }).catch((err: any) => {
-      console.error(err);
-    });
-  }
-
-  signIn = () => (!localStorage.getItem(MESSAGE_KEY)) ? this.signMessage() : this.authenticate();
-
-  syncAccount = () => {
-    const url = 
-    `${this._config.calimeroWebSdkService}/api/v1/shards/${this._config.shardId}/wallet/api/v1/account/sync`;
-
-    axios.post(
-      url,
-      {
-        calimeroToken: localStorage.getItem(CALIMERO_TOKEN_KEY),
-        secretToken: localStorage.getItem(AUTH_TOKEN_KEY)
-      }
-    ).then((res: any) => {
-      if (res.status == 200) {
-        return true;
-      }
-    }).catch((err: any) => {
-      console.error(err);
-      return false;
-    });
-  }
+  signIn = () => (!localStorage.getItem(MESSAGE_KEY)) && this.signMessage();
 
   signTransaction = (transactionString: string, callbackUrl: string) => {
     if(!this.isSignedIn) {
-      console.log("SignIn required before sign a transaction");
+      window.alert("SignIn required before sign a transaction");
```

Comment:
Too aggressive, return a result. Dapps should manage the output of this function. Alert is ok for the example dapp